std/algebra/emulated/fields_bls12381/e12_pairing.go

```diff
@@ -392,82 +166,59 @@ func (e Ext12) FrobeniusSquareTorus(y *E6) *E6 {
 //
 // [On Proving Pairings]: https://eprint.iacr.org/2024/640.pdf
 func (e Ext12) AssertFinalExponentiationIsOne(x *E12) {
```

Comment:
The method is already in this package, but wouldn't it make more sense to move it to `sw_bls12381` package? Imo rest of pairing related methods are exposed there (FinalExp, ML). I think that because how we have defined the `Pairing` type, then even if we call `Pairing.AssertFinalExponentiationIsOne` actually calls `Pairing.Ext12.AssertFinalExponentiationIsOne`.

But now looking at it, seems that we would have to refactor in all curve packages. Maybe it is not worth it and can keep as is (or do in separate PR).